x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/analytics/action/PutAnalyticsCollectionAction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.analytics.action;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.master.MasterNodeRequest;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.entsearch.analytics.AnalyticsCollection;
+import org.elasticsearch.xpack.entsearch.analytics.action.support.BaseAnalyticsCollectionResponse;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class PutAnalyticsCollectionAction extends ActionType<PutAnalyticsCollectionAction.Response> {
+
+    public static final PutAnalyticsCollectionAction INSTANCE = new PutAnalyticsCollectionAction();
+    public static final String NAME = "cluster:admin/behavioral_analytics/put";
+
+    public PutAnalyticsCollectionAction() {
+        super(NAME, PutAnalyticsCollectionAction.Response::new);
+    }
+
+    public static class Request extends MasterNodeRequest<Request> {
```

Comment:
Same here, why MasterNodeRequest?

Reply:
Leaning on @afoucret  to give a better answer here but we are using the MasterNodeRequest due to the way we use `clusterChanged` event to update the cached list of analytics collections in the `AnalyticsCollectionService`. From what I understand, we need to wait for master node acknowledgement so that the cluster metadata event has been already emitted and cache updated before we respond back.  

Happy to use the normal `ActionRequest` if you don't think this is right.

Reply:
Ok I see, then I wonder if we should change the AnalyticsCollectionService to be available on the master node only? @afoucret are we expecting the service to be available on every node?

Reply:
I believe it needs to be available on every node as we require the analyticsCollections cache to be used by the event intake API. We need to verify that the analytics collection exists before we can emit the event to the events log.